src/relcast.erl

```diff
@@ -1224,19 +1224,31 @@ do_serialize(Mod, Old, New, Prefix, Transaction) ->
                           K;
                      ({{K, V}, {_, OV}}) ->
                           KeyName = get_key_name(Prefix, K),
-                          case is_map(V) of
+                          case is_keytree_map(V) of
                               true ->
                                   do_serialize(K, fixup_old_map(OV), V, <<KeyName/binary, "_">>, Transaction);
-                              false ->
-                                  %% lager:info("writing ~p to disk", [K]),
+                              false when is_binary(V) ->
                                   ok = rocksdb:transaction_put(Transaction, KeyName, V),
+                                  K;
+                              false ->
+                                  %% some other term, try to term to binary as close to the edge
+                                  %% as possible to avoid the relcast modules
+                                  %% creating large binaries that have to be compared byte by byte
+                                  Encoded = term_to_binary(V, [compressed]),
```

Comment:
We should probably make the t2b options come from the app env, so they could be configured.

Reply:
done